target/linux/pistachio/base-files/lib/netifd/proto/lowpan.sh

```diff
@@ -19,9 +19,18 @@ proto_lowpan_setup() {
         local id=${ifname##*[[:alpha:]]}
         local interface="$cfg$id" # i.e wpan0
 
-        generate_mac $WPAN_OTP_REG0
-        local WPAN_EUI=$(echo $MAC | cut -d: -f1,2,3):ff:fe:$(echo $MAC | cut -d: -f4,5,6)
-
+        # Wifi mac address must be part of DTB to get access to it
+        local WIFI_STA_MAC=0x$(hexdump -n 6 -v -e '/1 "%02X"' /proc/device-tree/uccp@18480000/mac-address0)
+
+        # 6LOWPAN_MAC = WIFI_STA_MAC + 5
+        local LOWPAN_OTP_OFFSET=0x5
+        local LOWPAN_MAC=$(printf "0x%X" $((WIFI_STA_MAC + LOWPAN_OTP_OFFSET)))
+
+        # Compute EUI from MAC
+        local WPAN_EUI=$(printf "%012X" $((LOWPAN_MAC ^ 0x020000000000)))
+        WPAN_EUI=$(echo $WPAN_EUI | sed 's/.\{2\}/&:/g')        # Insert ':' every 2 characters
```

Comment:
Found a solution that doesn't need eval http://unix.stackexchange.com/questions/79349/convert-mac-address-to-link-local-address-with-bash